### DISCUSSION TITLE
Allow inline ref struct initialization for method parameter

### DISCUSSION THREAD

**Discussion by daningalla** [dotnet/csharplang#5702] [Language Ideas] [answered]
Consider the following:
```csharp
void ReadBytes(ref SequenceReader<byte> sequenceReader)
{
  // ...
}
```

The compiler disallows `ReadBytes(ref new SequenceReader<byte>(memory))`?

Would the compiler inferring a local scoped instance of the ref struct type be more than syntactic sugar?

**Comment by Joe4evr** [comment]
This is not exclusive to `ref struct`s; *any* method parameter that is passed by `ref`/`out` must be an lvalue.

**Comment by svick** [comment, marked as answer]
This is quite similar to https://github.com/dotnet/csharplang/issues/4355.

**Comment by daningalla** [comment]
Closing, thanks for the input
